### ISSUE TITLE
JEM4.2.1 on J!5.1 w/o compat plugin: JObject in editevent/editvenue

### ISSUE THREAD

**Issue by jojo12** [jemproject/JEM-Project#1726]
There seems to be a problem, when trying to replace JObject

**Comment by mckillo** [comment]
Tell me the file

**Comment by jojo12** [comment]
editevent and editvenue gives this error, It says for JObject use CMSObject, but this gives an error too

**Comment by jojo12** [comment]
# | Function | Location

1 | () | JROOT\libraries\vendor\joomla\utilities\src\ArrayHelper.php:70
2 | Joomla\Utilities\ArrayHelper::toObject() | JROOT\components\com_jem\models\editevent.php:94
3 | JemModelEditevent->getItem() | JROOT\components\com_jem\controllers\event.php:81
4 | JemControllerEvent->allowEdit() | JROOT\libraries\src\MVC\Controller\FormController.php:394
5 | Joomla\CMS\MVC\Controller\FormController->edit() | JROOT\components\com_jem\controllers\event.php:132
6 | JemControllerEvent->edit() | JROOT\libraries\src\MVC\Controller\BaseController.php:730
7 | Joomla\CMS\MVC\Controller\BaseController->execute() | JROOT\components\com_jem\jem.php:50
8 | require_once() | JROOT\libraries\src\Dispatcher\LegacyComponentDispatcher.php:71
9 | Joomla\CMS\Dispatcher\LegacyComponentDispatcher::Joomla\CMS\Dispatcher\{closure}() | JROOT\libraries\src\Dispatcher\LegacyComponentDispatcher.php:73
10 | Joomla\CMS\Dispatcher\LegacyComponentDispatcher->dispatch() | JROOT\libraries\src\Component\ComponentHelper.php:361
11 | Joomla\CMS\Component\ComponentHelper::renderComponent() | JROOT\libraries\src\Application\SiteApplication.php:218
12 | Joomla\CMS\Application\SiteApplication->dispatch() | JROOT\libraries\src\Application\SiteApplication.php:261
13 | Joomla\CMS\Application\SiteApplication->doExecute() | JROOT\libraries\src\Application\CMSApplication.php:306
14 | Joomla\CMS\Application\CMSApplication->execute() | JROOT\includes\app.php:58
15 | require_once() | JROOT\index.php:32

**Comment by jojo12** [comment]
for editvenue:
1 | () | JROOT\libraries\vendor\joomla\utilities\src\ArrayHelper.php:70
-- | -- | --
2 | Joomla\Utilities\ArrayHelper::toObject() | JROOT\components\com_jem\models\editvenue.php:88
3 | JemModelEditvenue->getItem() | JROOT\components\com_jem\controllers\venue.php:81
4 | JemControllerVenue->allowEdit() | JROOT\libraries\src\MVC\Controller\FormController.php:394
5 | Joomla\CMS\MVC\Controller\FormController->edit() | JROOT\components\com_jem\controllers\venue.php:121
6 | JemControllerVenue->edit() | JROOT\libraries\src\MVC\Controller\BaseController.php:730
7 | Joomla\CMS\MVC\Controller\BaseController->execute() | JROOT\components\com_jem\jem.php:50
8 | require_once() | JROOT\libraries\src\Dispatcher\LegacyComponentDispatcher.php:71
9 | Joomla\CMS\Dispatcher\LegacyComponentDispatcher::Joomla\CMS\Dispatcher\{closure}() | JROOT\libraries\src\Dispatcher\LegacyComponentDispatcher.php:73
10 | Joomla\CMS\Dispatcher\LegacyComponentDispatcher->dispatch() | JROOT\libraries\src\Component\ComponentHelper.php:361
11 | Joomla\CMS\Component\ComponentHelper::renderComponent() | JROOT\libraries\src\Application\SiteApplication.php:218
12 | Joomla\CMS\Application\SiteApplication->dispatch() | JROOT\libraries\src\Application\SiteApplication.php:261
13 | Joomla\CMS\Application\SiteApplication->doExecute() | JROOT\libraries\src\Application\CMSApplication.php:306
14 | Joomla\CMS\Application\CMSApplication->execute() | JROOT\includes\app.php:58
15 | require_once() | JROOT\index.php:32

**Comment by jojo12** [comment]
not yet fixed:
new error:
0 Joomla\CMS\Form\Field\EditorField::filter field `articletext` calls a deprecated filter class JComponentHelper, the class needs to be namespaced use \Joomla\CMS\Component\ComponentHelper instead or activate the backward compatible plugin. 
\src\Form\FormField.php:1102
-- | -- | --
2 | Joomla\CMS\Form\FormField->filter() | JROOT\libraries\src\Form\Form.php:1047
3 | Joomla\CMS\Form\Form->filter() | JROOT\libraries\src\Form\Form.php:991
4 | Joomla\CMS\Form\Form->process() | JROOT\libraries\src\MVC\Model\FormModel.php:219
5 | Joomla\CMS\MVC\Model\FormModel->validate() | JROOT\libraries\src\MVC\Controller\FormController.php:636
6 | Joomla\CMS\MVC\Controller\FormController->save() | JROOT\components\com_jem\controllers\event.php:292
7 | JemControllerEvent->save() | JROOT\libraries\src\MVC\Controller\BaseController.php:730
8 | Joomla\CMS\MVC\Controller\BaseController->execute() | JROOT\components\com_jem\jem.php:50
9 | require_once() | JROOT\libraries\src\Dispatcher\LegacyComponentDispatcher.php:71
10 | Joomla\CMS\Dispatcher\LegacyComponentDispatcher::Joomla\CMS\Dispatcher\{closure}() | JROOT\libraries\src\Dispatcher\LegacyComponentDispatcher.php:73
11 | Joomla\CMS\Dispatcher\LegacyComponentDispatcher->dispatch() | JROOT\libraries\src\Component\ComponentHelper.php:361
12 | Joomla\CMS\Component\ComponentHelper::renderComponent() | JROOT\libraries\src\Application\SiteApplication.php:218
13 | Joomla\CMS\Application\SiteApplication->dispatch() | JROOT\libraries\src\Application\SiteApplication.php:261
14 | Joomla\CMS\Application\SiteApplication->doExecute() | JROOT\libraries\src\Application\CMSApplication.php:306
15 | Joomla\CMS\Application\CMSApplication->execute() | JROOT\includes\app.php:58
16 | require_once() | JROOT\index.php:32

**Comment by mckillo** [comment]
Try to change in \components\com_jem\models\forms\event.xml line 6 to:
`               filter="\Joomla\CMS\Component\ComponentHelper::filterText" hide="pagebreak"/>`
I can't see the error, for me, it works.

**Comment by jojo12** [comment]
it's enough to write: 
`               filter="ComponentHelper::filterText" hide="pagebreak"/>`

**Comment by mckillo** [comment]
Yes, but in the other filters (venue and category XML file) are so with the path.
